.github/workflows/check_mergeability_ghstack.yml

```diff
@@ -1,29 +1,100 @@
-name: Check mergeability and dependencies for ghstack prs
+name: Check mergeability of ghstack PR
 
 on:
   pull_request:
     types: [opened, synchronize, reopened, edited]
 
 jobs:
-  check-regex:
+  ghstack-mergeability-check:
     runs-on: ubuntu-latest
-    outputs:
-      regex-match: ${{ steps.regex-match.outputs.match }}
     steps:
-      - uses: actions/checkout@v4
-
-      - id: regex-match
+      - id: check-if-ghstack
+        # checks whether the PR is a ghstack PR
         uses: actions-ecosystem/action-regex-match@d50fd2e7a37d0e617aea3d7ada663bd56862b9cc
         with:
           text: ${{ github.head_ref }}
           regex: '^(gh/[^/]+/[0-9]+/)head$'
 
-  pr-dependencies-check:
-    needs: check-regex
-    if: ${{ needs.check-regex.outputs.regex-match != '' }}
-    uses: pytorch/test-infra/.github/workflows/pr-dependencies-check.yml@main
-    with:
-      pr_number: ${{ github.event.pull_request.number }}
+      - name: Determine if should run
+        # checks whether the PR is a ghstack PR or whether PR changes the workflow files
+        id: should-run
+        shell: bash
+        run: |
+          # use GITHUB API to get the files changed in the PR
+          # and check if the following files were changed:
+          # .github/workflows/check_mergeability_ghstack.yml
+          # .github/scripts/trymerge.py
+          FILES_CHANGED=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
```

Comment:
This looks fine, but in case you need something more complicated, there are many GHA ready made to get the list of change files in a PR, i.e. https://github.com/tj-actions/changed-files

Reply:
I already tried that action :)

It's description is a bit confusing, what it actually does, is checking the locally modified files, rather than the changest of the PR.